sdk/storage/storage-blob/review/storage-blob.api.ts

```diff
@@ -0,0 +1,1424 @@
+  // WARNING: The type "Metadata" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobSetMetadataResponse" needs to be exported by the package (e.g. added to index.ts)
+  setMetadata(metadata?: Metadata, options?: BlobSetMetadataOptions): Promise<Models.BlobSetMetadataResponse>;
+  // WARNING: The type "Models.BlobSetTierResponse" needs to be exported by the package (e.g. added to index.ts)
+  setTier(tier: BlockBlobTier | PremiumPageBlobTier | string, options?: BlobSetTierOptions): Promise<Models.BlobSetTierResponse>;
+  // WARNING: The type "Models.BlobStartCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobStartCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  startCopyFromURL(copySource: string, options?: BlobStartCopyFromURLOptions): Promise<Models.BlobStartCopyFromURLResponse>;
+  // WARNING: The type "Models.BlobCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobCopyFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  syncCopyFromURL(copySource: string, options?: BlobSyncCopyFromURLOptions): Promise<Models.BlobCopyFromURLResponse>;
+  // WARNING: The type "Models.BlobUndeleteResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlobUndeleteResponse" needs to be exported by the package (e.g. added to index.ts)
+  undelete(options?: BlobUndeleteOptions): Promise<Models.BlobUndeleteResponse>;
+  withSnapshot(snapshot: string): BlobClient;
+}
+
+// @public
+interface BlobCreateSnapshotOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+  metadata?: Metadata;
+}
+
+// @public
+interface BlobDeleteOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+  deleteSnapshots?: Models.DeleteSnapshotsOptionType;
+}
+
+// @public
+interface BlobDownloadOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+  maxRetryRequests?: number;
+  progress?: (progress: TransferProgressEvent) => void;
+  rangeGetContentCrc64?: boolean;
+  rangeGetContentMD5?: boolean;
+  snapshot?: string;
+}
+
+// @public
+interface BlobGetPropertiesOptions {
+  abortSignal?: AbortSignalLike;
+  blobAccessConditions?: BlobAccessConditions;
+  customerProvidedKey?: Models.CpkInfo;
+}
+
+// @public
+interface BlobReleaseLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+interface BlobRenewLeaseOptions {
+  abortSignal?: AbortSignalLike;
+  modifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+class BlobSASPermissions {
```

Comment:
it's really not clear to me which structs get to have the parse/toString() pattern and which are kept as simple interfaces. I'm guessing the former are encoded chunks from the service and the latter are options the developer specifies to the API surface, but it feels a little arbitrary.

It seems like the encoding/decoding should all happen under the covers and the developer is only aware of interfaces. Unless the developer herself needs to ever stringify/JSON encode these things directly?